tsdb/engine/tsm1/engine.go

```diff
@@ -65,13 +66,13 @@ func NewDevEngine(path string, walPath string, opt tsdb.EngineOptions) tsdb.Engi
 		FileStore: fs,
 		Compactor: c,
 		CompactionPlan: &DefaultPlanner{
-			WAL:       w,
 			FileStore: fs,
-			Cache:     cache,
 		},
 		RotateFileSize:    DefaultRotateFileSize,
 		MaxFileSize:       MaxDataFileSize,
 		MaxPointsPerBlock: DefaultMaxPointsPerBlock,
+
+		MinCacheFlushThreshold: uint64(opt.Config.WALFlushMemorySizeThreshold),
```

Comment:
I think this config option should be `CacheFlushMemorySizeThreshold` now to be consistent with the `CacheMaxMemorySize` option.